### PR TITLE
Add request_header feature

### PR DESCRIPTION
**Pull request by nekitmish**
```
@winter.route_get('url/')
@winter.request_header('X-Auth-CompanyID', to='company_id')
def some_method(self, company_id: str):
  print(company_id)
```

### CODE REVIEW

File: winter/web/request_header_resolver.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+from winter.core.json import JSONDecodeException
+from winter.core.json import json_decode
+from winter.web.exceptions import RequestDataDecodeException
+from winter.web.request_header_annotation import RequestHeaderAnnotation
+
+
+class RequestHeaderArgumentResolver(ArgumentResolver):
+    def __init__(self):
+        super().__init__()
+        self._cache = {}
```
Comment (pristupa) [inline winter/web/request_header_resolver.py:17]: why cache?

File: winter/web/request_header_resolver.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        return any(annotation.map_to == argument.name for annotation in annotations)
+
+    def resolve_argument(
+        self,
+        argument: ComponentMethodArgument,
+        request: Request,
+        response_headers: MutableMapping[str, str],
+    ):
+        request_headers = request.headers
+        annotation = next(  # pragma: no cover
```
Comment (pristupa) [inline winter/web/request_header_resolver.py:30]: why no cover here?

**Review by pristupa — CHANGES_REQUESTED** [review]

File: winter/web/request_header_resolver.py (hunk 2 above)
Comment (nekitmish) [reply winter/web/request_header_resolver.py]: just forgot to delete

File: winter/web/request_header_resolver.py (hunk 1 above)
Comment (nekitmish) [reply winter/web/request_header_resolver.py]: just forgot to delete

**Review by pristupa — APPROVED** [review]

**Review by rushduhhp — APPROVED** [review]

**Pull request merged**